pkg/driver/controller_test.go

```diff
@@ -147,7 +147,9 @@ func TestCreateVolume(t *testing.T) {
 	stdVolCap := []*csi.VolumeCapability{
 		{
 			AccessType: &csi.VolumeCapability_Mount{
-				Mount: &csi.VolumeCapability_MountVolume{},
+				Mount: &csi.VolumeCapability_MountVolume{
+					FsType: "ext4",
+				},
```

Comment:
Suggest _not_ touching this variable, since you are implicitly modifying three dozen existing unit tests. If the new block size code can't handle `defaultFsType`, I think that's a bug that should be fixed, not worked around in unit tests.

But it is worthwhile to test both cases here (when fsType is both specified and not).

Reply:
The problem here is that the default fsType is de-facto set by the external-attacher: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/86f165e433eb7a7aff14a56540d9dd695dc1a756/charts/aws-ebs-csi-driver/templates/controller.yaml#L195

This means that if the `fsType` received for `CreateVolume` is blank I have no idea what `fsType` I'll be getting later when the volume is attached, and by then it will be too late.

No other code in `CreateVolume` touches the fstype so that's why I didn't bother making a separate test setup for it, but I can if you think it's relevant.

Reply:
> The problem here is that the default fsType is de-facto set by the external-attacher

Then I'm not sure how I feel about the `defaultFsType` [code](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/pkg/driver/node.go#L49) in _pkg/driver/node.go_. Confused, I guess. In any case, as long as the default supports block size configuration, everything should just work, right? I don't love the idea of baking that assumption into a unit test. But as assumptions go, that one seems pretty safe.

> No other code in CreateVolume touches the fstype **today**

FTFY 😄. Let's write unit tests that make the fewest possible assumptions about the code they're testing.

Reply:
> Then I'm not sure how I feel about the defaultFsType [code](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/pkg/driver/node.go#L49) in pkg/driver/node.go.

That code will handle the case if the volume is attached with a blank fstype. With kubernetes, that won't ever actually happen (because, as I mentioned above, the external attacher has its own default fstype logic), but it is possible with other COs.

We can move to assuming the default fstype will support a custom block size. Today, that's a relatively safe option, as it is highly unlikely anybody is using NTFS as their default filesystem, and that is the only filesystem for which custom block sizes is not supported.

Reply:
Because of the discussion above, the controller code will now allow blank fstypes (and thus the case when the default fstype is invalid will fail on mount), so this change is no longer necessary. 

I agree we should test a variety of capabilities here, but that's a relatively big change and out of scope for this PR - I will raise a PR at a later date to improve this testing.